src/main/java/com/jwplayer/southpaw/util/S3Helper.java

```diff
@@ -198,116 +266,128 @@ public void syncFromS3(URI localUri, URI s3Uri) throws InterruptedException {
+                        String suffix = StringUtils.substringAfter(getPath(localFile), getPath(localUri));
+                        boolean copyMe = true;
+                        for (S3ObjectSummary summary : summaries) {
+                            if (summary.getKey().endsWith(suffix) && localFile.lastModified() <= summary.getLastModified().getTime()) {
+                                copyMe = false;
+                                break;
+                            }
+                        }
+                        if (copyMe) localFilesToCopy.add(localFile);
+                    }
 
-        // Copy the new/updated files to S3
-        if(localFilesToCopy.size() > 0) {
-            TransferManager tx = TransferManagerBuilder.standard().withS3Client(s3).build();
-            List<Upload> uploads = new ArrayList<>();
-            for (File fileToCopy : localFilesToCopy) {
-                String fullPath = getPath(fileToCopy);
-                String suffix = StringUtils.substringAfter(fullPath, getPath(localUri));
-                String key = getPath(s3Uri) + suffix;
-                ObjectMetadata metadata = new ObjectMetadata();
-                metadata.setLastModified(new Date(fileToCopy.lastModified()));
-                PutObjectRequest request = new PutObjectRequest(bucket, key, fileToCopy).withMetadata(metadata);
-                uploads.add(tx.upload(request));
-                uploadCount++;
-                if(uploadCount % MAX_KEYS_PER_S3_OP == 0) {
-                    for(Upload upload: uploads) {
-                        upload.waitForUploadResult();
+                    // Copy the new/updated files to S3
+                    if (localFilesToCopy.size() > 0) {
+                        TransferManager tx = TransferManagerBuilder.standard().withS3Client(s3).build();
+                        List<Upload> uploads = new ArrayList<>();
+                        for (File fileToCopy : localFilesToCopy) {
+                            String fullPath = getPath(fileToCopy);
+                            String suffix = StringUtils.substringAfter(fullPath, getPath(localUri));
+                            String key = getPath(s3Uri) + suffix;
+                            ObjectMetadata metadata = new ObjectMetadata();
+                            metadata.setLastModified(new Date(fileToCopy.lastModified()));
+                            PutObjectRequest request = new PutObjectRequest(bucket, key, fileToCopy).withMetadata(metadata);
+                            uploads.add(tx.upload(request));
+                            uploadCount++;
+                            metrics.s3FilesUploaded.mark(1);
+                            if (uploadCount % MAX_KEYS_PER_S3_OP == 0) {
+                                for (Upload upload : uploads) {
+                                    upload.waitForUploadResult();
+                                }
+                                logger.info(String.format("Uploaded %s files", uploadCount));
```

Comment:
Same comment about this uploaded log message

Reply:
See above.